src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -260,7 +260,10 @@ static void notify_on_write(grpc_tcp* tcp) {
   if (grpc_tcp_trace.enabled()) {
     gpr_log(GPR_INFO, "TCP:%p notify_on_write", tcp);
   }
-  if (grpc_event_engine_run_in_background()) {
+
+  static bool event_engine_run_in_background =
+      grpc_event_engine_run_in_background();
+  if (event_engine_run_in_background) {
```

Comment:
I think this is going to compile down to a lock / call / unlock . Can this memoization be done in the plugin initialization instead?